utopia-remix/app/models/project.server.ts

```diff
@@ -168,30 +168,57 @@ export async function listSharedWithMeProjectsAndCollaborators(params: {
   projects: ProjectListing[]
   collaborators: CollaboratorsByProject
 }> {
-  // 1. grab the project IDs where the user is the collaborator,
-  // is not an owner, are not deleted, and have the COLLABORATIVE access.
-  const projects = await prisma.project.findMany({
+  // 1. grab the projects for which the user is a collaborator, are not deleted, are
+  // collaborative, and the user is not the owner.
+  const projectsAndCollaborators = await prisma.projectCollaborator.findMany({
     where: {
-      ProjectAccess: { access_level: AccessLevel.COLLABORATIVE },
-      ProjectCollaborator: { some: { user_id: params.userId } },
-      owner_id: { not: params.userId },
-      OR: [{ deleted: null }, { deleted: false }],
+      user_id: params.userId, // the user is a collaborator
+      Project: {
+        OR: [{ deleted: null }, { deleted: false }], // project is not deleted
+        owner_id: { not: params.userId }, // the owner is not the user
+        ProjectAccess: { access_level: AccessLevel.COLLABORATIVE }, // the project is collaborative
+      },
     },
     select: {
-      ...selectProjectWithoutContent,
-      ProjectCollaborator: { include: { User: true } },
+      Project: {
+        select: {
+          ...selectProjectWithoutContent,
+          ProjectCollaborator: { select: { User: true } },
+        },
+      },
     },
-    orderBy: { modified_at: 'desc' },
+    orderBy: { Project: { modified_at: 'desc' } },
+  })
+  const projects = projectsAndCollaborators.map(({ Project }) => Project)
+
+  // 2. grab the owner details in case some are missing from the collaborators
```

Comment:
> in case some are missing from the collaborators

The owner details will always be missing from the collaborators, as they have been explicitly excluded from the query above

Reply:
Oh sorry, I realise this is referencing the next step